iroh-gateway/src/core.rs

```diff
@@ -396,6 +396,42 @@ mod tests {
         assert!(body.starts_with(b"<!DOCTYPE html>"));
     }
 
+    #[tokio::test]
+    async fn test_raw_fetch() {
+        let files = &[(
+            "hello.txt".to_string(),
+            Alphanumeric
+                .sample_string(&mut SmallRng::seed_from_u64(42), 8 * 1024 * 1024)
+                .bytes()
+                .collect(),
+        )];
+        let large_file = &files[0].1;
+        let test_setup = setup_test(false, files).await;
+
+        let res = do_request(
+            "GET",
+            &format!("localhost:{}", test_setup.gateway_addr.port()),
+            &format!("/ipfs/{}/hello.txt", test_setup.root_cid),
+            None,
+        )
+        .await;
+        assert_eq!(http::StatusCode::OK, res.status());
+        let body = hyper::body::to_bytes(res.into_body()).await.unwrap();
+        assert_eq!(&body[..], large_file);
+
+        let res = do_request(
+            "GET",
+            &format!("localhost:{}", test_setup.gateway_addr.port()),
+            &format!("/ipfs/{}/hello.txt?format=raw", test_setup.root_cid),
+            None,
+        )
+        .await;
+        assert_eq!(http::StatusCode::OK, res.status());
+        let body = hyper::body::to_bytes(res.into_body()).await.unwrap();
+        let ufs = iroh_unixfs::unixfs::UnixfsNode::decode(&test_setup.root_cid, body).unwrap();
```

Comment:
Am slightly confused on how to unpack this further without significantly complicating things.